test/allowlist_for_publicAPI.json

```diff
@@ -393,7 +393,8 @@
     "set_debug_level_from_env",
     "timedelta",
     "ProcessGroupMPI",
-    "ProcessGroupNCCL"
+    "ProcessGroupNCCL",
+    "ProcessGroupUCC"
```

Comment:
No new entries should be added to this file.
Please make sure that only the functions that should be public are actually public.